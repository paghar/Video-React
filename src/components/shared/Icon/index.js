//@flow
import React from 'react';
import { IconSpan } from './index.style';

type Props = {
  iconNumber: number,
  fontSize: string,
  color: string,
  hoverColor?: string,
  selectable?: boolean,
  clicked?: () => void,
  className?: string,
};

function Icon({
  fontSize,
  color,
  iconNumber,
  hoverColor,
  selectable,
  clicked,
  className,
}: Props) {
  return (
    <IconSpan
      fontSize={fontSize}
      iconColor={color}
      hoverColor={hoverColor}
      selectable={selectable}
      className={className}
      onClick={clicked}
    >
      {String.fromCharCode(parseInt(iconNumber))}
    </IconSpan>
  );
}

export default Icon;
