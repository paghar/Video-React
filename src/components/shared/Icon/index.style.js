import styled from 'styled-components';

export const IconSpan = styled.span`
  font-family: 'FontAwesome' !important;
  font-size: ${({ fontSize }) => fontSize};
  color: ${({ iconColor }) => iconColor};
  margin: 0px;
  padding: 0px;
  align-self: center;
  ${({ selectable }) =>
    !selectable &&
    `
         -moz-user-select: none;
         -khtml-user-select: none;
         -webkit-user-select: none;
         -ms-user-select: none;
         user-select: none;
        `}
  :hover {
    color: ${({ hoverColor }) => hoverColor};
  }
`;
