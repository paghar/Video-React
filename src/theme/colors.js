import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
