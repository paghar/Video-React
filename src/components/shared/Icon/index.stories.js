import React from 'react';
import { Secondary } from 'theme/colors';
import Story from '.';

export default { title: 'Icon' };
export const IconWithFontSize20px = () => (
  <Story fontSize="20px" color={Secondary.iii} iconNumber="61449"></Story>
);
export const IconWithDeferentColor = () => (
  <Story fontSize="12px" color={Secondary.iv} iconNumber="61614"></Story>
);

export const IconWithHover = () => (
  <Story
    fontSize="24px"
    color={Secondary.iv}
    hoverColor={Secondary.iii}
    iconNumber="61614"
  ></Story>
);

export const IconIsNotSelectable = () => (
  <Story
    fontSize="24px"
    color={Secondary.iv}
    hoverColor={Secondary.iii}
    iconNumber="61614"
    selectable={true}
  ></Story>
);
