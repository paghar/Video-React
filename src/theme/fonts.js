import theme from 'styled-theming';

export const Title1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '22px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '35px',
  }),
};

export const Title2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '21px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '33px',
  }),
};

export const Display1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display1M = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display3 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display4 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '12px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '20px',
  }),
};
export const Display5 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
export const Display6 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
