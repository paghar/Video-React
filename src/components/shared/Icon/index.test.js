//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '.';
import { Secondary } from 'theme/colors';

describe('<Icon />', () => {
  it('Icon with 20px size', () => {
    const tree = renderer
      .create(
        <Icon fontSize="20px" color={Secondary.iii} iconNumber={61449}></Icon>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Icon with 12px size ', () => {
    const tree = renderer
      .create(
        <Icon fontSize="12px" color={Secondary.iv} iconNumber={61614}></Icon>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Icon with hover ', () => {
    const tree = renderer
      .create(
        <Icon
          fontSize="24px"
          color={Secondary.iv}
          hoverColor={Secondary.iii}
          iconNumber={61614}
        ></Icon>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Selectable Icon', () => {
    const tree = renderer
      .create(
        <Icon
          fontSize="24px"
          color={Secondary.iv}
          hoverColor={Secondary.iii}
          iconNumber={61614}
          selectable={true}
        ></Icon>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
