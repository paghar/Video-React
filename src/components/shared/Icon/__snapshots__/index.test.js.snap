// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Icon with 12px size  1`] = `
.c0 {
  font-family: 'FontAwesome' !important;
  font-size: 12px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<span
  className="c0"
  fontSize="12px"
>
  
</span>
`;

exports[`<Icon /> Icon with 20px size 1`] = `
.c0 {
  font-family: 'FontAwesome' !important;
  font-size: 20px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<span
  className="c0"
  fontSize="20px"
>
  
</span>
`;

exports[`<Icon /> Icon with hover  1`] = `
.c0 {
  font-family: 'FontAwesome' !important;
  font-size: 24px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<span
  className="c0"
  fontSize="24px"
>
  
</span>
`;

exports[`<Icon /> Selectable Icon 1`] = `
.c0 {
  font-family: 'FontAwesome' !important;
  font-size: 24px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<span
  className="c0"
  fontSize="24px"
>
  
</span>
`;
